### PR TITLE
tests: add UT for catalogs

### LINKED ISSUE #2919: Add unit tests for catalog
**Summary**

https://github.com/datafuselabs/databend/tree/ac2164eac76b6c3c12d694127fc5d47f25b96000/query/src/catalogs

### PR DESCRIPTION
**Pull request by sunisdown**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Add UT for catalog

## Changelog

- Build/Testing/CI

## Related Issues

Fixes #2919

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by sunisdown** [comment]
@BohuTANG  I ran the test locally and it passed, how can I manually retrigger the CI ?

**Comment by PsiACE** [comment]
I manually re-run the ci, good luck

**Comment by BohuTANG** [comment]
Looks it's `thread_local` issue?

**Comment by BohuTANG** [comment]
@mergify update

**Comment by BohuTANG** [comment]
```
---- catalogs::database_catalog::test_catalogs_table stdout ----
thread 'catalogs::database_catalog::test_catalogs_table' panicked at 'assertion failed: `(left == right)`
  left: `1`,
 right: `2`', query/tests/it/catalogs/database_catalog.rs:83:5
stack backtrace:
```

**Comment by sunisdown** [comment]
> ```
> ---- catalogs::database_catalog::test_catalogs_table stdout ----
> thread 'catalogs::database_catalog::test_catalogs_table' panicked at 'assertion failed: `(left == right)`
>   left: `1`,
>  right: `2`', query/tests/it/catalogs/database_catalog.rs:83:5
> stack backtrace:
> ```

Will check it  later :) thanks

File: query/tests/it/catalogs/database_catalog.rs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+use common_meta_types::DropDatabaseReq;
+use common_meta_types::DropTableReq;
+use common_meta_types::TableMeta;
+use databend_query::catalogs::Catalog;
+
+use crate::tests::create_catalog;
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn test_catalogs_get_database() -> Result<()> {
+    let catalog = create_catalog().unwrap();
```
Comment (BohuTANG) [inline query/tests/it/catalogs/database_catalog.rs:35]: It would be better change 'unwrap' -> '?' to catch the result instead of panic :)

Comment (sunisdown) [reply query/tests/it/catalogs/database_catalog.rs]: Morning, updated

**Comment by BohuTANG** [comment]
/lgtm

Thank you @sunisdown

**Pull request merged**